lib/semantics/check-do-concurrent.cc

```diff
@@ -204,6 +249,20 @@ class DoConcurrentLabelEnforce {
   void Post(const parser::ErrLabel &errLabel) { checkLabelUse(errLabel.v); }
   void Post(const parser::EndLabel &endLabel) { checkLabelUse(endLabel.v); }
   void Post(const parser::EorLabel &eorLabel) { checkLabelUse(eorLabel.v); }
+  void Post(const parser::DoConstruct &d) { do_depth_--; }
+  void checkName(const std::optional<parser::Name> &nm) {
+    if (!nm.has_value()) {
+      if (do_depth_ == 0) {
+        messages_.Say(currentStatementSourcePosition_,
+            "exit from DO CONCURRENT construct"_err_en_US);
+      }
+      return;
```

Comment:
Might be more clear if written as an `else` rather than a `return`.

Reply:
Made this change.